noise-research/secure-transfer.md

```diff
@@ -0,0 +1,243 @@
+
+
+- The device `A` sends a commitment to its static key `sA` because:
+    - `A` cannot adaptively choose a static key based on the state of the Noise handshake at the end of message `1`, i.e. after the authentication code is confirmed. Note that device `A` is trusted in our security assumptions.
+    - Confirming the authentication code after processing message `1` will ensure that no MitM can send a static key different than `sA`.
+
+- The authorization code is shown and has to be confirmed at the end of message `1` because:
+    - an attacker that frontruns device `A` by sending faster his own ephemeral key would be detected before  he's able to know device `B` static key `sB`;
+    - it ensures that no MitM attacks will happen during *the whole* pairing handshake, since commitments to the (later exchanged) device static keys will be implicitly acknowledged by the authorization code confirmation;
+    - it enables to safely swap the role of handshake initiator and responder (see above);
+
+- Device `B` sends his static key first because:
+    - by being the pairing requester, it cannot probe device `A` identity without revealing its own (static key) first. Note that device `B` static key and its commitment can be binded to other cryptographic material (e.g., seed phrase).
+
+- Device `B` opens a commitment to its static key at message `2.` because:
+    - if device `A` replies concluding the handshake according to the protocol, device `B` acknowledges that device `A` correctly received his static key `sB`, since `r` was encrypted under an encryption key derived from the static key `sB` and the genuine (due to the previous `authcode` verification) ephemeral keys `eA` and `eB`.
+
+- Device `A` opens a commitment to its static key at message `3.` because:
+    - if device `B` doesn't abort the pairing, device `A` acknowledges that device `B` correctly received his static key `sA`, since `s` was encrypted under an encryption key derived from the static keys `sA` and `sB` and the genuine (due to the previous `authcode` verification) ephemeral keys `eA` and `eB`.
+
+# Secure Transfer
+
+Once the handshake is concluded, sensitive information can be exchanged using the encryption keys agreed during the pairing phase. If stronger security guarantees are required, some [additional tweaks](#Additional-Possible-Tweaks) are possible.
+
+In the following subsections we report the details of applications which are currently under development, mainly in order to implement [35/WAKU2-SESSION]().
+
+However, the pairing and transfer phases descriptions are designed to be application-agnostic, and should be flexible enough to mutually authenticate and allow secure communication of two devices over a distributed network of Waku2 nodes.
+
+## N11M session management mechanism
+
+In this scenario, one of Alice's devices is already communicating with one of Bob's devices within an active Noise session, e.g. after a successful execution of a Noise handshake.
```

Comment:
By `after a successful execution of a Noise handshake.` do you mean a pairing process or not?